guiEditor/src/components/propertyTab/propertyTabComponent.tsx

```diff
@@ -152,59 +152,67 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
+                        if (snippet.version && snippet.version != "0") {
+                            adt.snippetId += "#" + snippet.version;
+                        }
+                        const windowAsAny = window as any;
+                        if (windowAsAny.Playground && oldId) {
+                            windowAsAny.Playground.onRequestCodeChangeObservable.notifyObservers({
+                                regex: new RegExp(oldId, "g"),
+                                replace: `parseFromSnippetAsync("${adt.snippetId}`,
+                            });
+                        }
+                        resolve(adt.snippetId);
+                    } else {
+                        reject("string");
                     }
-
-                    const windowAsAny = window as any;
-
-                    if (windowAsAny.Playground && oldId) {
-                        windowAsAny.Playground.onRequestCodeChangeObservable.notifyObservers({
-                            regex: new RegExp(oldId, "g"),
-                            replace: `parseFromSnippetAsync("${adt.snippetId}`,
-                        });
-                    }
-
-                    alert("GUI saved with ID: " + adt.snippetId + " (please note that the id was also saved to your clipboard)");
-                } else {
-                    alert("Unable to save your GUI");
                 }
-            }
-        };
+            };
+
+            xmlHttp.open("POST", AdvancedDynamicTexture.SnippetUrl + (adt.snippetId ? "/" + adt.snippetId : ""), true);
+            xmlHttp.setRequestHeader("Content-Type", "application/json");
+            const dataToSend = {
+                payload: JSON.stringify({
+                    gui: content,
+                }),
+                name: "",
+                description: "",
+                tags: "",
+            };
+            xmlHttp.send(JSON.stringify(dataToSend));
+        });
+    }
+    
 
-        xmlHttp.open("POST", AdvancedDynamicTexture.SnippetUrl + (adt.snippetId ? "/" + adt.snippetId : ""), true);
-        xmlHttp.setRequestHeader("Content-Type", "application/json");
+    saveToSnippetServer = async () => {
+        const adt = this.props.globalState.guiTexture;
+        const content = JSON.stringify(adt.serializeContent());
 
-        const dataToSend = {
-            payload: JSON.stringify({
-                gui: content,
-            }),
-            name: "",
-            description: "",
-            tags: "",
-        };
+        const savePromise = this.props.globalState.customSave?.action || this.saveToSnippetServerHelper;
+        savePromise(content, adt).then((resolve: string) => {
+            adt.snippetId = resolve;
+            if (navigator.clipboard) {
+                navigator.clipboard.writeText(adt.snippetId);
+            }
+            alert("GUI saved with ID: " + adt.snippetId + " (please note that the id was also saved to your clipboard)");

```

Comment:
Are we using alert in GE to communicate with the user ???

I guess a message panel like sandbox or playground could be better

Reply:
what @msDestiny14 said was that NME uses the same alert. We had this conversation before, and i believe UX-wise alerts are bad, but we will need a form of dialog or local alert/toast system to avoid using alerts.